### PR TITLE
drivers: sensor: lis2dh: convert to spi_dt_spec

### PR DESCRIPTION
**Pull request by bbilas**
Convert lis2dh driver to use `spi_dt_spec` helpers.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

File: drivers/sensor/lis2dh/lis2dh.h [hunk 1]
```diff
@@ -173,20 +173,13 @@ union lis2dh_sample {
...
-#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
-
 union lis2dh_bus_cfg {
 #if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
 	uint16_t i2c_slv_addr;
 #endif
 
 #if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
-	const struct lis2dh_spi_cfg *spi_cfg;
+	struct spi_dt_spec spi;
```
Comment (avisconti) [inline drivers/sensor/lis2dh/lis2dh.h:182]: Did you drop 'const' on purpose?

**Review by avisconti — CHANGES_REQUESTED** [review]
LGTM, just explain why dropping 'const'.

Thanks for the change!

File: drivers/sensor/lis2dh/lis2dh.h (hunk 1 above)
Comment (gmarull) [reply drivers/sensor/lis2dh/lis2dh.h:182]: @avisconti it's no longer a pointer

Comment (avisconti) [reply drivers/sensor/lis2dh/lis2dh.h:182]: > @avisconti it's no longer a pointer

Yes, correct. But the structure is constant indeed and cannot be changed at runtime (only assigned at compile time from DT). Isn't it?

Comment (gmarull) [reply drivers/sensor/lis2dh/lis2dh.h:182]: but the whole driver config is already `const`, right?

Comment (avisconti) [reply drivers/sensor/lis2dh/lis2dh.h:182]: yes, right

**Review by avisconti — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

File: drivers/sensor/lis2dh/lis2dh.h (hunk 1 above)
Comment (bbilas) [reply drivers/sensor/lis2dh/lis2dh.h:182]: Sorry for my late answer. @gmarull explained that quite :)

**Pull request merged**